mailpoet/lib/Validator/Validator.php

```diff
@@ -25,11 +25,28 @@ public function __construct(
    * @return mixed
    */
   public function validate(Schema $schema, $value, string $paramName = 'value') {
-    $result = $this->validateAndSanitizeValueFromSchema($value, $schema->toArray(), $paramName);
-    if ($result instanceof WP_Error) {
-      throw ValidationException::createFromWpError($result);
+
+    $schema = $schema->toArray();
+    if (!isset($schema['properties'])) {
+      $result = $this->validateAndSanitizeValueFromSchema($value, $schema, $paramName);
+      if ($result instanceof WP_Error) {
+        throw ValidationException::createFromWpError($result);
+      }
+      return $result;
+    }
+
+    $combinedError = new WP_Error();
+    foreach ($schema['properties'] as $propertyKey => $definition) {
```

Comment:
I think this should actually go outside of `mailpoet/lib/Validator/Validator.php` somewhere in the automation code, only reusing the validator from outside.

The reason here is that objects can have some validation rules on the whole object itself, but here we're breaking it  (what I mean is things like `minProperties`, etc.). Or... what do you think?